gamerun.cc

```diff
@@ -0,0 +1,47 @@
+#include "gamerun.h"
+using namespace std;
+
+/*
+TO DO:
+- Add pickMove() method to Human
+- Create Levelx subclasses for Computer, each with a pickMove() method
+*/
+
+int gameRun(Observer *white, Observer *black, ChessGame *gameBoard) {
+    int turn = gameBoard->getplayerTurn();
+    int toReturn = INT_MAX;
+    string comm = "";
+    string a = "";
+    string b = "";
+
+    while (toReturn == INT_MAX) {
+        if (turn == 1) {
+            cout << "White's Turn: ";
+            bool resignStatus = white->pickMove();       //so pickMove() will have bool return type
+            if (resignStatus == 1) {                    //never executed with compcomp
+                toReturn = -1;
+            } else if (gameBoard->isStalemate() == 1) {
+                toReturn = 0;
+            } else if (gameBoard->isBlackKingChecked() == 1) {
+                toReturn = 1;
+            } else {
+                gameBoard->switchTurn();
+                turn = gameBoard->getplayerTurn();
+            }
+        } else {
+            cout << "Black's Turn: ";
+            bool resignStatus  = black->pickMove();
+            if (resignStatus == 1) {
+                toReturn = 1;
+            } else if (gameBoard->isStalemate == 1) {
+                 toReturn = 0;
+            } else if (gameBoard->isWhiteKingChecked() == 1) {
+                toReturn = -1;
+            } else {
+                gameBoard->switchTurn();
+                turn = gameBoard->getplayerTurn();
+            }
+        }
+    }
+    return toReturn;
```

Comment:
is it possible for this function to return any other value aside from 0, 1, or -1? It seems like the only way to reach this statement is if the loop in this file is broken. And the only way for the loop to be broken is if `toReturn` is equal to 0, 1, or -1.

Reply:
toReturn returning 0, 1 or -1 denotes a successful game i.e. a game that has gone till completion. I intend to account for the error of a game being left unfinished in which case INT_MAX will be returned. I haven't done that yet because it's not asked of us, but if I get the time I will.